### ISSUE TITLE
Bug: ReadConfig silently fails if ConfigType not set

### ISSUE THREAD

**Issue by mmindenhall** [spf13/viper#109]
This is pretty minor, but a bug nonetheless.  As a workaround for #90, I switched from using `ReadInConfig` to using `ReadConfig`:

``` go
    // old code
    config.SetConfigName("my_config")
    config.AddConfigPath(configPath)
    err = config.ReadInConfig()
    if err != nil {
        log.Fatalln("Error reading config file: ", err)
    }
```

``` go
    // new code
    f, err := os.Open(configPath + "/my_config.json")
    if err != nil {
        log.Fatalln("Error opening config file: ", err.Error())
    }
    r := JsonConfigReader.New(f)  // JsonConfigReader strips comments and trailing commas
    err = config.ReadConfig(r)
    if err != nil {
        log.Fatalln("Error reading config file: ", err.Error())
    }
```

`ReadInConfig` automatically detects the config type as "json" from the config file extension.  However, the new code above was silently failing to read the config, as the `ConfigType` had not been set, and the code falls all the way through the switch statement in `marshallConfigReader` without doing anything.  It seems there should be a check for `ConfigType` in `ReadConfig`, or a `default:` that returns an error in the `marshallConfigReader` switch statement.

**Comment by carldunham** [comment]
Crickets...

**Comment by dexterlb** [comment]
This issue is still relevant and incredibly annoying. It would even be better for `ReadConfig()` to panic instead of failing silently.

**Comment by sagikazarmark** [comment]
Fixed in #1857
